src/entities/course.ts

```diff
@@ -29,18 +29,23 @@ export type DisplayCourse = {
 export const displayCourseToApi = (
   displayedCourse: Partial<DisplayCourse> &
     Pick<DisplayCourse, "registeredCourse" | "absence" | "attendance" | "late">,
-  methods: MethodJa[] = []
-): RegisteredCourse => ({
-  ...displayedCourse.registeredCourse,
-  attendance: displayedCourse.attendance,
-  absence: displayedCourse.absence,
-  late: displayedCourse.late,
-  memo: displayedCourse.memo ?? "",
-  name: displayedCourse.name ?? "",
-  instructor: displayedCourse.instructor ?? "",
-  schedules: scheduleToApi(displayedCourse.schedules ?? []),
-  methods: methods.map(jaToMethod),
-});
+  methods?: MethodJa[]
+): RegisteredCourse => {
+  const course = {
+    ...displayedCourse.registeredCourse,
+    attendance: displayedCourse.attendance,
+    absence: displayedCourse.absence,
+    late: displayedCourse.late,
+    memo: displayedCourse.memo ?? "",
+    name: displayedCourse.name ?? "",
+    instructor: displayedCourse.instructor ?? "",
+    schedules: scheduleToApi(displayedCourse.schedules ?? []),
+  };
+  if (methods != null) {
+    course.methods = methods.map(jaToMethod);
+  }
```

Comment:
これできない？
```suggestion
    methods: methods?.map(jaTpMethod) ?? [],
  };
```
条件分岐はできるだけ避けたい

Reply:
これだとmethodが空の配列で上書きされませんか？

Reply:
もしもベース講義にあった場合、methodsは値を持つ。
ベース講義がない、もしくは授業形式が登録されていない場合、空なので`[]`で上書きしても影響はでない

Reply:
apiToDisplayCourse
```
method: blankToChar(
    (registeredCourse.methods ?? registeredCourse.course?.methods ?? [])
      .map(methodToJa)
      .join(", "),
    char
  ),
```
`registeredCourse.methods`が`null`で、詳細画面からupdateした場合、`registeredCourse.methods`が`[]`になってしまい、勝手に`[]`に上書きされてしまうことになる？

Reply:
> 命題1 registeredCourse.methodsが`[] or null`のとき、ベース講義も`[] or null`である。
> 命題2 ベース講義のmethodsに何かしらの値(v)がある、場合updateでregisteredCourse.methodsもその値(v)で登録される

その挙動自体に不具合が発生しないのなら、一行にまとめたいと思いました